game-core/src/main/java/games/strategy/triplea/image/ResourceImageFactory.java

```diff
@@ -81,4 +85,23 @@ private JPanel getResourcesPanel(final ResourceCollection resources, final boole
     return resourcePanel;
   }
 
+  /**
+   * Returns button with resource amounts and given text. If resources is empty then returns
+   * button with just the text.
+   */
+  public JButton getResourcesButton(final ResourceCollection resources, final String text) {
+    if (resources.isEmpty()) {
+      return new JButton(text);
+    }
+    final JPanel panel = getResourcesPanel(resources);
+    panel.add(new JLabel(text));
+    panel.setSize(panel.getPreferredSize());
+    panel.doLayout();
+    final BufferedImage image = new BufferedImage(panel.getWidth(), panel.getHeight(), BufferedImage.TYPE_INT_RGB);
+    final Graphics2D g = image.createGraphics();
+    panel.paint(g);
```

Comment:
Actually, I think this is a pretty creative solution given that `JButton` doesn't support rendering an arbitrary `JComponent` on its face like many other GUI frameworks do.

However, the difference between the panel background and the button background is a bit annoying.  It looks like you can fix that with a two-line change:

```diff
diff --git a/game-core/src/main/java/games/strategy/triplea/image/ResourceImageFactory.java b/game-core/src/main/java/games/strategy/triplea/image/ResourceImageFactory.java
index c9edde172..f7d38fd05 100644
--- a/game-core/src/main/java/games/strategy/triplea/image/ResourceImageFactory.java
+++ b/game-core/src/main/java/games/strategy/triplea/image/ResourceImageFactory.java
@@ -94,10 +94,11 @@ public class ResourceImageFactory extends AbstractImageFactory {
       return new JButton(text);
     }
     final JPanel panel = getResourcesPanel(resources);
+    panel.setOpaque(false);
     panel.add(new JLabel(text));
     panel.setSize(panel.getPreferredSize());
     panel.doLayout();
-    final BufferedImage image = new BufferedImage(panel.getWidth(), panel.getHeight(), BufferedImage.TYPE_INT_RGB);
+    final BufferedImage image = new BufferedImage(panel.getWidth(), panel.getHeight(), BufferedImage.TYPE_INT_ARGB);
     final Graphics2D g = image.createGraphics();
     panel.paint(g);
     g.dispose();
```

That produces the following example (my mouse is over the Finland button):

![user-actions](https://user-images.githubusercontent.com/4826349/46355386-b4dd5700-c62e-11e8-9ca3-c7f48ce064c7.png)

Reply:
Nice find. Definitely makes it look a lot nicer.